### ISSUE TITLE
How to use the disparity image to calculate real distances.

### ISSUE THREAD

**Issue by DiJester** [nianticlabs/monodepth2#94]
I think disparity image only contains relative depth information but not absolute distance, right?

**Comment by aj96** [comment]
@DiJester You can compute metric distances if you replace the PoseCNN with ground truth camera-to-camera transformation across time-steps. You can compute that transformation for all examples using the provided GPS/IMU data in kitti dataset (oxts folder). I describe the process here: https://github.com/nianticlabs/monodepth2/issues/78 The trick was to disable the random horizontal flipping and take out examples with noisy GPS/IMU data. Some of the examples had the altitude changing by 5 meters in one time-step.

**Comment by mdfirman** [comment]
@DiJester , thanks for your interest here!

You are right that models trained with *monocular only* training data do not predict distances in real-world metric. However, models trained with *stereo* or *mono+stereo* training data do make real-world metric predictions in metres.

Even in the mono case, the model is _reasonably_ consistent in its scale between test images, so you can find a single scaling factor to turn predictions into metric scale (see the paper for more details on this).

**Comment by DiJester** [comment]
Thank you for response.
